Upgrade to PostGIS 3.5.0

elprans opened the pull request.

elprans committed:
Upgrade to PostGIS 3.5.0

msullivan commented:
(I think this is probably the last extension package upgrade we can do without having an upgrade script)

vpetrovykh committed:
Tweak the extraction process.

Update which SQL files need to be introspected for generating the EdgeQL for
the extension. Clean up the template, since the `<>` is now properly
implemented and we don't need to fake it.

Remove debug print from the tests.

Include the generated .rst in the repo (so that it can be potentially
used by the docs).

vpetrovykh reviewed (approved):
I've updated the scripts and  templates. I've changed where the gen script looks for SQL files (that need to be built) and included the comments file (which is now separate). The template no longer needs to add `<>` since it is implemented properly in the SQL extension.
